paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/orphan/FlinkOrphanFilesClean.java

```diff
@@ -202,31 +238,86 @@ public void endInput() throws IOException {
                                 });
 
         usedFiles = usedFiles.union(usedManifestFiles);
-
-        List<String> fileDirs =
-                listPaimonFileDirs().stream()
-                        .map(Path::toUri)
-                        .map(Object::toString)
-                        .collect(Collectors.toList());
-        DataStream<Pair<String, Long>> candidates =
-                env.fromCollection(fileDirs)
+        FileStorePathFactory pathFactory = table.store().pathFactory();
+        List<Tuple7<String, String, String, String, String, Integer, String>> tablePaths =
+                Arrays.asList(
+                        new Tuple7<>(
+                                table.fullName(),
+                                pathFactory.manifestPath().toString(),
+                                pathFactory.indexPath().toString(),
+                                pathFactory.statisticsPath().toString(),
+                                pathFactory.dataFilePath().toString(),
+                                partitionKeysNum,
+                                table.store().options().dataFileExternalPaths()));
+        DataStream<Tuple2<String, Long>> candidates =
+                env.fromCollection(
+                                tablePaths,
+                                TypeInformation.of(
+                                        new TypeHint<
+                                                Tuple7<
+                                                        String,
+                                                        String,
+                                                        String,
+                                                        String,
+                                                        String,
+                                                        Integer,
+                                                        String>>() {}))
```

Comment:
Same as above. since tablePaths.size() is 1, a bigger parallelism would be a waste of resource.